modules/tinymce/src/themes/silver/main/ts/ui/core/color/Options.ts

```diff
@@ -1,13 +1,26 @@
 import { Transformations } from '@ephox/acid';
-import { Arr, Type } from '@ephox/katamari';
+import { Arr, Obj, Type } from '@ephox/katamari';
 
 import Editor from 'tinymce/core/api/Editor';
 import { EditorOptions } from 'tinymce/core/api/OptionTypes';
 import { Menu } from 'tinymce/core/api/ui/Ui';
 
 import { ColorCache } from './ColorCache';
 
-const colorCache = ColorCache(10);
+const foregroundId = 'forecolor';
+const backgroundId = 'hilitecolor';
+
+interface CacheStorage {
```

Comment:
I think this belongs in a CacheStore.ts module then the state getCacheForId, addColor could all live in there instead of this Options.ts. Separation of concerns.